test_resultify.py

```diff
@@ -54,4 +54,32 @@ def test_filter_fuel(self):
 
         index = ['REGION', 'TIMESLICE', 'TECHNOLOGY', 'FUEL', 'YEAR']
 
+        pd.testing.assert_frame_equal(actual.set_index(index), expected.set_index(index), check_index_type=False)
+    
+class TestEnergy:
+
+    def test_filter_primary_energy(self):
+        filepath = os.path.join("tests","fixtures","ProductionByTechnologyAnnual.csv")
+        input_data = pd.read_csv(filepath)
+
+        technologies = ['****00I00','****00X00','**OC*****','**SO*****','**WI*****']
```

Comment:
You might want to split this test up into multiple tests, one for each pattern which is matched?  What happens if there's an incorrect pattern passed in. For example `*********` or `**I*AM*TOO*LONG`?

Reply:
Hint: to check that an error is raised in Pytest, you use the following:
```python
with raises(ValueError) as ex:
    filter_primary_energy(input_data, bogus_data)
assert str(ex) == "The error message which is raised"
```

Reply:
Thanks for the tips! I will work on this.

Reply:
Well now the input are regex. Not sure how to test if a regex makes sense. Any suggestion?